src/sys/socket/mod.rs

```diff
@@ -527,12 +529,36 @@ pub struct linger {
     pub l_linger: c_int
 }
 
+/// Kernel credential structure.
+///
+/// This structure is equivalent to [libc::ucred].
+///
+/// [libc::ucred]: http://www.man7.org/linux/man-pages/man7/unix.7.html
+#[cfg(all(target_os = "linux", not(target_arch="arm")))]
 #[repr(C)]
 #[derive(Clone, Copy, PartialEq, Eq, Debug)]
-pub struct ucred {
-    pid: pid_t,
-    uid: uid_t,
-    gid: gid_t,
+pub struct UserCredentials {
+    pid: Pid,
+    uid: Uid,
+    gid: Gid,
+}
+
+#[cfg(all(target_os = "linux", not(target_arch="arm")))]
+impl UserCredentials {
+    /// Get the process ID of the sending process.
+    pub fn pid(&self) -> Pid {
+        self.pid
+    }
+
+    /// Get the user ID of the sending process.
+    pub fn user(&self) -> Uid {
```

Comment:
In terms of naming, there is first `pid`, 
but then `user` and `group`, instead of `uid` and `gid`.
Won't this be confusing for the user (who would for instance expect `user` to be a string)?